security/advancedtls/pemfile_provider_test.go

```diff
@@ -0,0 +1,220 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package advancedtls
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/credentials/tls/certprovider"
+	"google.golang.org/grpc/security/advancedtls/testdata"
+)
+
+func restoreMockFunctionsBack() {
+	readKeyCertPairFunc = tls.LoadX509KeyPair
+	readTrustCertFunc = func(trustFile string) (*x509.CertPool, error) {
+		trustData, err := ioutil.ReadFile(trustFile)
+		if err != nil {
+			return nil, err
+		}
+		trustPool := x509.NewCertPool()
+		ok := trustPool.AppendCertsFromPEM(trustData)
+		if !ok {
+			return nil, fmt.Errorf("failed to call AppendCertsFromPEM")
+		}
+		return trustPool, nil
+	}
+}
+
+func (s) TestNewPEMFileProvider(t *testing.T) {
+	// We deliberately use an empty watching function for watching goroutine, since we are not testing watching behaviors.
+	tests := []struct {
+		desc          string
+		certFile      string
```

Comment:
You can define a single field `options PEMFileProviderOptions` here instead of three fields. And when initializing the options, only specify the fields which contain a value. No need to specify fields which are going to be empty.